src/Core/src/Platform/Android/ImageButtonExtensions.cs

```diff
@@ -29,29 +29,9 @@ public static async void UpdatePadding(this ShapeableImageView platformButton, I
 
 			// Because there is only a single padding property, we need to reset the padding to 0 otherwise we
 			// probably will get a double padding. Trust me. I've seen it happen. It's not pretty.
+			// The padding is also reset in MauiShapeableImageView.
 			platformButton.SetPadding(0, 0, 0, 0);
 
-			// The padding has a few issues, but setting and then resetting after some calculations
-			// are done seems to work. This is a workaround for the following issue:
-			// https://github.com/material-components/material-components-android/issues/2063
-			await Task.Yield();
-
-			if (!platformButton.IsAlive())
-				return;
-
-			// We must re-set all the paddings because the first time was not hard enough.
-			platformButton.SetContentPadding((int)padding.Left, (int)padding.Top, (int)padding.Right, (int)padding.Bottom);
-			platformButton.SetPadding(0, 0, 0, 0);
-
-			// Just like before, the bugs are not done. This needs to trigger a re-calculation of
```

Comment:
@mattleibow You have added this part (ShapeAppearanceModel) in #22298. I am not sure if I can just remove it. None of the tests that looked relevant failed on my machine.

Reply:
I am scared tbh. I forget the corner cases where this happens - and they yield is especially scary. I think I was going off the fixes the native side was doing.

The current test set may not be enough and we will have to go remove this and see what breaks manually, and then make your change to see if it fixes it.